From c68ee22f73afb7038c5b5394bb944583beb809c4 Mon Sep 17 00:00:00 2001
From: Anton Jastsuk <anton@leil.io>
Date: Wed, 11 Mar 2026 14:01:16 +0200
Subject: [PATCH 1/2] build: Fix build failure with GCC 15 due to missing
 cstdint

In GCC 15 some C++ standard library headers no longer
include other headers that were being used internally by
the library. As such, C++ programs that used standard
library components without including the right headers
will no longer compile.
https://gcc.gnu.org/gcc-15/porting_to.html#header-dep-changes

Signed-off-by: Anton Jastsuk <anton@leil.io>
---
 src/common/compact_vector.h     | 1 +
 src/common/memory_mapped_file.h | 1 +
 2 files changed, 2 insertions(+)

diff --git a/src/common/compact_vector.h b/src/common/compact_vector.h
index a1214f3d..0629ddc3 100644
--- a/src/common/compact_vector.h
+++ b/src/common/compact_vector.h
@@ -23,6 +23,7 @@
 
 #include <algorithm>
 #include <cassert>
+#include <cstdint>
 #include <iterator>
 #include <limits>
 #include <memory>
diff --git a/src/common/memory_mapped_file.h b/src/common/memory_mapped_file.h
index b5530e8d..3af59618 100644
--- a/src/common/memory_mapped_file.h
+++ b/src/common/memory_mapped_file.h
@@ -18,6 +18,7 @@
 
 #include "common/platform.h"
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
-- 
2.53.0

